### PR TITLE
add blog post about jenkins job builder

### PR DESCRIPTION
**Pull request by kdelee**
Short demo of using docker to serve up local instance of Jenkins for development of jenkins-job-builder generated jobs.

Please let me know if there are any edits you would like me to make!

### CODE REVIEW

**Comment by kdelee** [comment]
You can see it rendered on my fork: ~https://kdelee.github.io/pulpproject.org/2017/07/05/jenkins-job-builder/~
https://kdelee.github.io/pulpproject.org/2017/07/10/jenkins-job-builder/

File: _posts/2017-07-05-jenkins-job-builder.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+only to experience a hard drive failure to lose all your work, or much more
+likely, something changes in your set up that leads you to hand updating dozens
+of jobs so carefully configured with hours of mouse clicks.
+
+Jenkins Job Builder (JJB)
+----------------
+
+To scale the process of Jenkins job definition it become obvious that one must
+be able to generate Jenkins jobs based on templates and with variable
+replacement so that you might test multiple versions against multiple OS's, and
```
Comment (daviddavis) [inline _posts/2017-07-05-jenkins-job-builder.md]: s/OS's/OSes/

File: _posts/2017-07-05-jenkins-job-builder.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+more over that these assets should be under version control. The good news is,
```
Comment (daviddavis) [inline _posts/2017-07-05-jenkins-job-builder.md]: s/more over/moreover,/

File: _posts/2017-07-05-jenkins-job-builder.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+PulpQE stores the YAML files and scripts used to generate the pulp jenkins jobs
+with jenkins-job-builder in the
+[pulp_packaging](https://github.com/pulp/pulp_packaging) repo. 
+
+While working on changes to these files, it is desirable to be able to try
+them out locally, so I decided to give it a shot!
+
+I have [Docker](https://www.docker.com/) set up on my Fedora 25 machine, so I spun up a Jenkins instance to
+test it out: There is an official [Jenkins image on
```
Comment (daviddavis) [inline _posts/2017-07-05-jenkins-job-builder.md]: I would probably change the colon here to a period maybe?

File: _posts/2017-07-05-jenkins-job-builder.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+DockerHub](https://hub.docker.com/_/jenkins/) so I pulled it down and served it
+up on my localhost. _(More detailed guides for setting up jenkins from this
+docker image are available on the Jenkins DockerHub page)_
+
+    docker pull jenkins
+    docker run -p 8080:8080 -p 50000:50000 jenkins
+
+I then visited the Jenkins web ui at <code>http://localhost:8080</code> and was directed to
```
Comment (daviddavis) [inline _posts/2017-07-05-jenkins-job-builder.md]: s/ui/UI/

File: _posts/2017-07-05-jenkins-job-builder.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+title: Introduction to Jenkins Job Builder
+author: Elijah DeLee
+tags:
+  - pulpqe
+---
+
+The power of Jenkins
+----------------
+
+I first was exposed to Jenkins when I was working for a open source software
```
Comment (dralley) [inline _posts/2017-07-05-jenkins-job-builder.md]: "while working for" might sound better than "when I was working for"

File: _posts/2017-07-05-jenkins-job-builder.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+and [Semaphore](https://semaphoreci.com/) can be
+excellent if you can accomplish your build within the constraints of the
+systems they provide, they could not meet the needs of that project. Enter
+[Jenkins](https://jenkins.io/), "the leading open source automation server".
+
+Jenkins is an powerful tool, and much of this power comes from the fact that it
+is so incredibly customizable. Configuring jobs via the web UI that Jenkins
+provides leaves many things to be desired, however. First, the task is fraught
+with human error. It is very easy to forget to expand that one nested menu and
+set a critical value, perhaps creating a faulty job that leads to misleading
```
Comment (dralley) [inline _posts/2017-07-05-jenkins-job-builder.md]: This sentence should probably be broken up a bit into two or three sentences

File: _posts/2017-07-05-jenkins-job-builder.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+DockerHub](https://hub.docker.com/_/jenkins/) so I pulled it down and served it
+up on my localhost. _(More detailed guides for setting up jenkins from this
+docker image are available on the Jenkins DockerHub page)_
+
+    docker pull jenkins
+    docker run -p 8080:8080 -p 50000:50000 jenkins
+
+I then visited the Jenkins web ui at <code>http://localhost:8080</code> and was directed to
+login with an admin account created during the install with a temporary
+password created printed to the logs. I created a new admin account and password and chose to
```
Comment (daviddavis) [inline _posts/2017-07-05-jenkins-job-builder.md]: A bit of a run-on sentence. How about:

I then visited the Jenkins web UI at <code>http://localhost:8080</code> and was directed to login with an admin account created during the install. A temporary admin password could be found in the logs. 

Also, I'd probably drop "and password" here too. Too many ands.

File: _posts/2017-07-05-jenkins-job-builder.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+work down instead of speeding it up (that's what automation is _supposed_ to
+do, right?). Secondly, once you have constructed your house of cards, you have
+only to experience a hard drive failure to lose all your work, or much more
+likely, something changes in your set up that leads you to hand updating dozens
+of jobs so carefully configured with hours of mouse clicks.
+
+Jenkins Job Builder (JJB)
+----------------
+
+To scale the process of Jenkins job definition it become obvious that one must
```
Comment (dralley) [inline _posts/2017-07-05-jenkins-job-builder.md]: s/become/became

This sentence needs to be broken up a bit or rewritten to be shorter/multiple sentences.

**Review by daviddavis — CHANGES_REQUESTED** [review]
Looks great @kdelee. 👍 

A few small comments.

**Review by dralley — COMMENTED** [review]
Thank you for this PR @kdelee !

This was very good, I only had a few nitpicks but no serious objections

File: _posts/2017-07-05-jenkins-job-builder.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
+---
+title: Introduction to Jenkins Job Builder
```
Comment (daviddavis) [inline _posts/2017-07-05-jenkins-job-builder.md]: I might title this with something more specific like "Testing Jenkins Configurations with Jenkins Job Builder"

**Comment by kdelee** [comment]
@daviddavis @dralley 

Thanks for the review, I have pushed those changes to the branch on this PR and you should be able to see them on my fork!

**Review by daviddavis — APPROVED** [review]

**Review by dralley — APPROVED** [review]

**Pull request merged**